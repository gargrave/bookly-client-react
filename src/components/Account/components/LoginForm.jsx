import React from 'react'
import { func, shape, string } from 'prop-types'

const LoginForm = props => {
  return (
    <form noValidate style={{ maxWidth: '500px', margin: 'auto', marginTop: '20px', textAlign: 'left' }}>
      <div className="input-single">
        <label htmlFor="email">Email:</label>
        <input type="email" name="email" id="email" value={props.user.email} onChange={props.handleInputChange} />
      </div>

      <div className="input-single">
        <label htmlFor="password">Password:</label>
        <input
          type="password"
          name="password"
          id="password"
          value={props.user.password}
          onChange={props.handleInputChange}
        />
      </div>

      <div className="input-single">
        <button type="submit" onClick={props.handleLogin}>
          Login
        </button>
      </div>
    </form>
  )
}

LoginForm.propTypes = {
  user: shape({
    email: string.isRequired,
    password: string.isRequired
  }).isRequired,
  handleInputChange: func.isRequired,
  handleLogin: func.isRequired
}

export default LoginForm
