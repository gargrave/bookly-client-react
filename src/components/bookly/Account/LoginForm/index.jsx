import React from 'react'
import { func, shape, string } from 'prop-types'

import { buildClasses } from '@/utils/cssHelpers'

import '@/components/common/Form/styles.css'

const LoginForm = (props) => {
  return (
    <form
      className={buildClasses(['form', 'login-form'])}
      onSubmit={props.handleSubmit}
      noValidate
    >
      <div className="input-field">
        <label htmlFor="email">Email:</label>
        <input type="email" name="email" id="email" value={props.user.email} onChange={props.handleInputChange} />
      </div>

      <div className="input-field">
        <label htmlFor="password">Password:</label>
        <input
          type="password"
          name="password"
          id="password"
          value={props.user.password}
          onChange={props.handleInputChange}
        />
      </div>

      <div className="input-field">
        <button type="submit" className="submit-button" onClick={props.handleSubmit}>
          Login
        </button>
      </div>
    </form>
  )
}

LoginForm.propTypes = {
  user: shape({
    email: string.isRequired,
    password: string.isRequired,
  }).isRequired,
  handleInputChange: func.isRequired,
  handleSubmit: func.isRequired,
}

export default LoginForm
