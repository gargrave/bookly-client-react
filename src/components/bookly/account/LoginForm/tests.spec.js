import React from 'react'
import { shallow } from 'enzyme'

import LoginForm from './'

describe('LoginForm', () => {
  let props
  let component

  describe('with "user" populated', () => {
    beforeEach(() => {
      props = {
        user: { email: 'whatever@gmail.com', password: 'password' },
        handleInputChange: jest.fn(),
        handleSubmit: jest.fn(),
      }

      component = shallow(<LoginForm {...props} />)
    })

    test('renders correctly', () => {
      expect(component).toMatchSnapshot()
      expect(component.find('form.login-form').length).toEqual(1)
      expect(component.find('input[type="email"]').length).toEqual(1)
      expect(component.find('input[type="password"]').length).toEqual(1)
      expect(component.find('button.submit-button').length).toEqual(1)
    })

    test('populates the "email" field correctly', () => {
      const text = component.find('input[name="email"]').props().value
      expect(text).toEqual(props.user.email)
    })

    test('populates the "password" field correctly', () => {
      const text = component.find('input[name="password"]').props().value
      expect(text).toEqual(props.user.password)
    })

    test('calls the "handleSubmit" callback when the form is submitted', () => {
      expect(props.handleSubmit.mock.calls.length).toEqual(0)
      component.simulate('submit')
      expect(props.handleSubmit.mock.calls.length).toEqual(1)
    })

    test('calls the "handleSubmit" callback when the button is clicked', () => {
      expect(props.handleSubmit.mock.calls.length).toEqual(0)
      component.find('button.submit-button').simulate('click')
      expect(props.handleSubmit.mock.calls.length).toEqual(1)
    })

    test('calls the "handleInputChange" callback when "email" value changes', () => {
      const beforeLen = props.handleInputChange.mock.calls.length
      component.find('input[name="email"]').simulate('change')
      expect(props.handleInputChange.mock.calls.length).toEqual(beforeLen + 1)
    })

    test('calls the "handleInputChange" callback when "password" value changes', () => {
      const beforeLen = props.handleInputChange.mock.calls.length
      component.find('input[name="password"]').simulate('change')
      expect(props.handleInputChange.mock.calls.length).toEqual(beforeLen + 1)
    })
  })

  describe('with empty "user" data', () => {
    beforeEach(() => {
      props = {
        user: { email: '', password: '' },
        handleInputChange: jest.fn(),
        handleSubmit: jest.fn(),
      }

      component = shallow(<LoginForm {...props} />)
    })

    test('leaves the "email" input field blank', () => {
      const text = component.find('input[name="email"]').props().value
      expect(text.length).toEqual(0)
    })

    test('leaves the "password" input field blank', () => {
      const text = component.find('input[name="password"]').props().value
      expect(text.length).toEqual(0)
    })
  })
})
