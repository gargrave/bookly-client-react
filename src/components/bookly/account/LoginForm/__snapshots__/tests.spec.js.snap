// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm with "user" populated renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <form
    className="bookly-form bookly-login-form"
    noValidate={true}
    onSubmit={[Function]}
>
    <div
        className="input-field"
    >
        <label
            htmlFor="email"
        >
            Email:
        </label>
        <input
            id="email"
            name="email"
            onChange={[Function]}
            type="email"
            value="whatever@gmail.com"
        />
    </div>
    <div
        className="input-field"
    >
        <label
            htmlFor="password"
        >
            Password:
        </label>
        <input
            id="password"
            name="password"
            onChange={[Function]}
            type="password"
            value="password"
        />
    </div>
    <div
        className="input-field"
    >
        <button
            className="submit-button"
            onClick={[Function]}
            type="submit"
        >
            Login
        </button>
    </div>
</form>,
  "nodes": Array [
    <form
      className="bookly-form bookly-login-form"
      noValidate={true}
      onSubmit={[Function]}
>
      <div
            className="input-field"
      >
            <label
                  htmlFor="email"
            >
                  Email:
            </label>
            <input
                  id="email"
                  name="email"
                  onChange={[Function]}
                  type="email"
                  value="whatever@gmail.com"
            />
      </div>
      <div
            className="input-field"
      >
            <label
                  htmlFor="password"
            >
                  Password:
            </label>
            <input
                  id="password"
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value="password"
            />
      </div>
      <div
            className="input-field"
      >
            <button
                  className="submit-button"
                  onClick={[Function]}
                  type="submit"
            >
                  Login
            </button>
      </div>
</form>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <LoginForm
        handleInputChange={[Function]}
        handleSubmit={[Function]}
        user={
                Object {
                        "email": "whatever@gmail.com",
                        "password": "password",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "handleInputChange": [Function],
          "handleSubmit": [Function],
          "user": Object {
            "email": "whatever@gmail.com",
            "password": "password",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <form
          className="bookly-form bookly-login-form"
          noValidate={true}
          onSubmit={[Function]}
>
          <div
                    className="input-field"
          >
                    <label
                              htmlFor="email"
                    >
                              Email:
                    </label>
                    <input
                              id="email"
                              name="email"
                              onChange={[Function]}
                              type="email"
                              value="whatever@gmail.com"
                    />
          </div>
          <div
                    className="input-field"
          >
                    <label
                              htmlFor="password"
                    >
                              Password:
                    </label>
                    <input
                              id="password"
                              name="password"
                              onChange={[Function]}
                              type="password"
                              value="password"
                    />
          </div>
          <div
                    className="input-field"
          >
                    <button
                              className="submit-button"
                              onClick={[Function]}
                              type="submit"
                    >
                              Login
                    </button>
          </div>
</form>,
        "_debugID": 2,
        "_renderedOutput": <form
          className="bookly-form bookly-login-form"
          noValidate={true}
          onSubmit={[Function]}
>
          <div
                    className="input-field"
          >
                    <label
                              htmlFor="email"
                    >
                              Email:
                    </label>
                    <input
                              id="email"
                              name="email"
                              onChange={[Function]}
                              type="email"
                              value="whatever@gmail.com"
                    />
          </div>
          <div
                    className="input-field"
          >
                    <label
                              htmlFor="password"
                    >
                              Password:
                    </label>
                    <input
                              id="password"
                              name="password"
                              onChange={[Function]}
                              type="password"
                              value="password"
                    />
          </div>
          <div
                    className="input-field"
          >
                    <button
                              className="submit-button"
                              onClick={[Function]}
                              type="submit"
                    >
                              Login
                    </button>
          </div>
</form>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LoginForm
    handleInputChange={[Function]}
    handleSubmit={[Function]}
    user={
        Object {
            "email": "whatever@gmail.com",
            "password": "password",
          }
    }
/>,
}
`;
